golang/example/graph/comment.go

```diff
@@ -235,76 +237,36 @@ type searchCommentInput struct {
 	UpdatedAt *graphql.Time `json:"updated_at"`
 }
 
-// QueryMods returns a list of QueryMod based on the struct values
-func (s *searchCommentInput) QueryMods() []qm.QueryMod {
-	mods := []qm.QueryMod{}
-	// Get reflect value
-	v := reflect.ValueOf(s).Elem()
-	// Iterate struct fields
-	for i := 0; i < v.NumField(); i++ {
-		field := v.Type().Field(i) // StructField
-		value := v.Field(i)        // Value
-		if value.IsNil() || !value.IsValid() {
-			// Skip if field is nil
-			continue
-		}
-
-		// Get column name from tags
-		column, hasColumnName := field.Tag.Lookup("json")
-		// Skip if no DB definition
-		if !hasColumnName {
-			continue
-		}
-
-		operator := "="
-		val := value.Elem().Interface()
-		if dataType := field.Type.String(); (dataType == "string" || dataType == "*string") &&
-			val.(string) != "" {
-			operator = "LIKE"
-			val = fmt.Sprintf("%%%s%%", val)
-		}
-		mods = append(mods, qm.And(fmt.Sprintf("%s %s ?", column, operator), val))
-	}
-	return mods
-}
-
-// AllComments retrieves Comments based on the provided search parameters
-func (r *Resolver) AllComments(ctx context.Context, args struct {
-	Since    *graphql.ID
-	PageSize *int32
-	Search   *searchCommentInput
+// SearchComments retrieves Comments based on the provided search parameters
+func (r *Resolver) SearchComments(ctx context.Context, args struct {
+	PageNumber *int32
+	PageSize   *int32
+	Input      *searchCommentInput
 }) (CommentsCollection, error) {
 	result := CommentsCollection{}
 
-	pageSize := defaultPageSize // Default page size
-	if args.PageSize != nil {
-		pageSize = int(*args.PageSize)
-	}
 	mods := []qm.QueryMod{
-		qm.Limit(pageSize),
 		// TODO: Add eager loading based on requested fields
 		qm.Load("Post"),
 	}
 
-	if args.Since != nil {
-		s := string(*args.Since)
-		i, err := strconv.ParseInt(s, 10, 64)
-		if err != nil {
-			return result, err
-		}
-		mods = append(mods, qm.Offset(int(i)))
-	}
-	if args.Search != nil {
-		mods = append(mods, args.Search.QueryMods()...)
+	// Pagination
+	mods = append(mods, QueryModPagination(args.PageNumber, args.PageSize)...)
+
+	// Search input
+	if args.Input != nil {
+		mods = append(mods, QueryModsSearch(args.Input)...)
```

Comment:
wondering if it is a tinsy more elegant as

``` go
mods = append(mods, QueryModPagination(args.PageNumber, args.PageSize)...)
mods = append(mods, QueryModsSearch(args.Input)...) // could append empty
```

also, `QueryModSearch` suffice? without the `s` ?